client/src/components/controls/LineChart.vue

```diff
@@ -85,95 +84,92 @@ export default {
   methods: {
     initialize() {
       d3.select(this.$el)
-        .select("svg")
+        .select('svg')
         .remove();
-      var width = this.$el.clientWidth;
-      var height = this.$el.clientHeight;
-      var x = d3
+      let tooltipTimeoutHandle = null;
+      const tooltip = d3
+        .select(this.$el)
+        .append('div')
+        .attr('class', 'tooltip')
+        .style('display', 'none');
+      const width = this.$el.clientWidth;
+      const height = this.$el.clientHeight;
+      const x = d3
         .scaleLinear()
         .domain([this.startFrame, this.endFrame])
         .range([0, width]);
       this.x = x;
-      var max = d3.max(this.lineData, datum => d3.max(datum.values, d => d[1]));
-      var y = d3
+      const max = d3.max(this.lineData, (datum) => d3.max(datum.values, (d) => d[1]));
+      const y = d3
         .scaleLinear()
         .domain([0, Math.max(max + max * 0.2, 2)])
         .range([height, 0]);
 
-      var line = d3
+      const line = d3
         .line()
         .curve(d3.curveStepAfter)
-        .x(d => x(d[0]))
-        .y(d => y(d[1]));
+        .x((d) => x(d[0]))
+        .y((d) => y(d[1]));
       this.line = line;
 
-      var svg = d3
+      const svg = d3
         .select(this.$el)
-        .append("svg")
-        .style("display", "block")
-        .attr("width", width)
-        .attr("height", height)
-        .append("g")
-        .attr("transform", `translate(0,-1)`);
+        .append('svg')
+        .style('display', 'block')
+        .attr('width', width)
+        .attr('height', height)
+        .append('g')
+        .attr('transform', 'translate(0,-1)');
 
-      var axis = d3.axisRight(y).tickSize(width);
+      const axis = d3.axisRight(y).tickSize(width);
       this.axis = axis;
       svg
-        .append("g")
-        .attr("class", "axis-y")
+        .append('g')
+        .attr('class', 'axis-y')
         .call(axis)
-        .call(g =>
-          g
-            .selectAll(".tick text")
-            .attr("x", 0)
-            .attr("dx", 13)
-        );
+        .call((g) => g
+          .selectAll('.tick text')
+          .attr('x', 0)
+          .attr('dx', 13));
 
-      var path = svg
+      const path = svg
         .selectAll()
         .data(this.lineData)
         .enter()
-        .append("path")
-        .attr("class", "line")
-        .attr("d", d => line(d.values))
-        .style("stroke", d => (d.color ? d.color : "#4c9ac2"))
-        .on("mouseenter", function(d) {
-          var [x, y] = d3.mouse(this);
+        .append('path')
+        .attr('class', 'line')
+        .attr('d', (d) => line(d.values))
+        .style('stroke', (d) => (d.color ? d.color : '#4c9ac2'))
+        .on('mouseenter', (d) => {
+          const [_x, _y] = d3.mouse(this);
```

Comment:
Using the arrow function here rebinds `this` to the Vue Component instead of the current function.  Causing some errors to be logged because it is trying to find the position using a VueComponent as and argument for `d3.mouse(this)` instead of an svg element:
![image](https://user-images.githubusercontent.com/61746913/79262246-04752380-7e5f-11ea-8349-0f71bcc94d14.png)
We could change it back to utilizing a `function (d) {` or instead utilize `d3.event.currentTarget` although that has some warnings with Babel and Webpack.  It has the possibility of changing the d3.event during the event (source: [d3.event](https://github.com/d3/d3-selection#event))